### ISSUE TITLE
Gemspec/RequireMFA autocorrect does not detect some existing setting of metadata

### ISSUE THREAD

**Issue by mvz** [rubocop/rubocop#10251]
## Expected behavior

Given a gemspec containing lines like the following:
```ruby
  spec.metadata["source_code_uri"] = some_uri
```

Gemspec/RequireMFA should autocorrect by setting just `spec.metadata["rubygems_mfa_required"]`.

## Actual behavior

The autocorrect adds the following lines:
```ruby
  spec.metadata = {
    "rubygems_mfa_required" => "true"
  }
```
Which means the existing metadata entries are removed.

## Steps to reproduce the problem

1. Create a file `foo.gemspec` containing:
```ruby
# frozen_string_literal: true

Gem::Specification.new do |spec|
  spec.metadata['source_code_uri'] = 'https://foo.example.com/'
end
```
2. Create a file `.rubocop.yml` containing:
```yml
AllCops:
  NewCops: enable
```
3. Run `rubocop -a`

`foo.gemspec` now contains:
```ruby
# frozen_string_literal: true

Gem::Specification.new do |spec|
  spec.metadata['source_code_uri'] = 'https://foo.example.com/'
  spec.metadata = {
    'rubygems_mfa_required' => 'true'
  }
end
```

## RuboCop version

```
$ [bundle exec] rubocop -V
1.23.0 (using Parser 3.0.2.0, rubocop-ast 1.13.0, running on ruby 3.0.2 x86_64-linux)
```

**Comment by dvandersluis** [comment]
Thanks for catching this!
